### ISSUE TITLE
ES6: Use class Syntax, lack of check for constructor

### ISSUE THREAD

**Issue by 9064911** [freeCodeCamp/freeCodeCamp#17871]
#### Describe your problem and - if possible - how to reproduce it
Test for this lesson may not check the syntax for any constructor syntax, instead looks like it's tested by output.

EIther user provide the constructor(correct/expected way) or not, challenge passes.

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function/

https://forum.freecodecamp.org/t/some-bug-use-class-syntax-to-define-a-constructor-function/208378
#### Tell us about your browser and operating system

* Browser Name:  Chrome
* Browser Version: 67
* Operating System: Win7

**Comment by HadrienAllemon** [comment]
I'm on it.

**Comment by ZokiMar** [comment]
following solution works to passs the test: 

function makeClass() {
  "use strict";
  /* Alter code below this line */
class Vegetable {
  constructor(Vegetable) {
    this. Vegetable = Vegetable;
  }
}
  /* Alter code above this line */
  return Vegetable;
}
const Vegetable = makeClass();
const carrot = new Vegetable('carrot');
console.log(carrot.name); // => should be 'carrot'

**Comment by drzsolttorok** [comment]
Hello,
Unfortunately, this solution doesn't work for me. 
(The same as in the "Hint":
https://guide.freecodecamp.org/certifications/javascript-algorithms-and-data-structures/es6/use-class-syntax-to-define-a-constructor-function/
)
Error message:

// running tests
carrot.name should return carrot.
// tests completed

**Comment by scissorsneedfoodtoo** [comment]
@drzsolttorok, sorry to hear that! Would you mind posting your solution here so we could take a look?

**Comment by drzsolttorok** [comment]
Hello Kristofer,
Thank you for your email. (ES6: Use class Syntax to Define a Constructor
Function)
I have already got help from FreeCodeCamp.
I used the solution from "HINT", but it didn't work.

 class Vegetable {
     constructor(Vegetable){
       this.Vegetable = Vegetable;

     }
   }


What worked is the code below:

class Vegetable { constructor(name) { this.name = name; } }

Thank you.
Kind regards,
Zsolt
